sympy/assumptions/handlers/calculus.py

```diff
@@ -3,232 +3,205 @@
-        for arg in expr.args:
-            _bounded = ask(Q.finite(arg), assumptions)
-            if _bounded:
-                continue
-            elif _bounded is None:
-                if result is None:
-                    return None
-                if ask(Q.nonzero(arg), assumptions) is None:
-                    return None
-                if result is not False:
-                    result = None
-            else:
-                result = False
-        return result
-
-    @staticmethod
-    def Pow(expr, assumptions):
-        """
-        Unbounded ** NonZero -> Unbounded
-        Bounded ** Bounded -> Bounded
-        Abs()<=1 ** Positive -> Bounded
-        Abs()>=1 ** Negative -> Bounded
-        Otherwise unknown
-        """
-        base_bounded = ask(Q.finite(expr.base), assumptions)
-        exp_bounded = ask(Q.finite(expr.exp), assumptions)
-        if base_bounded is None and exp_bounded is None:  # Common Case
-            return None
-        if base_bounded is False and ask(Q.nonzero(expr.exp), assumptions):
-            return False
-        if base_bounded and exp_bounded:
-            return True
-        if (abs(expr.base) <= 1) == True and ask(Q.positive(expr.exp), assumptions):
-            return True
-        if (abs(expr.base) >= 1) == True and ask(Q.negative(expr.exp), assumptions):
-            return True
-        if (abs(expr.base) >= 1) == True and exp_bounded is False:
-            return False
-        return None
+                result = None
+        else:
+            result = False
+    return result
+
+@FinitePredicate.register(Pow)
+def _(expr, assumptions):
+    """
+    * Unbounded ** NonZero -> Unbounded
 
-    @staticmethod
-    def log(expr, assumptions):
-        return ask(Q.finite(expr.args[0]), assumptions)
+    * Bounded ** Bounded -> Bounded
 
-    exp = log
+    * Abs()<=1 ** Positive -> Bounded
 
-    cos, sin, Number, Pi, Exp1, GoldenRatio, TribonacciConstant, ImaginaryUnit, sign = \
-        [staticmethod(CommonHandler.AlwaysTrue)]*9
+    * Abs()>=1 ** Negative -> Bounded
 
-    Infinity, NegativeInfinity = [staticmethod(CommonHandler.AlwaysFalse)]*2
+    * Otherwise unknown
+    """
+    base_bounded = ask(Q.finite(expr.base), assumptions)
+    exp_bounded = ask(Q.finite(expr.exp), assumptions)
+    if base_bounded is None and exp_bounded is None:  # Common Case
+        return None
+    if base_bounded is False and ask(Q.nonzero(expr.exp), assumptions):
+        return False
+    if base_bounded and exp_bounded:
+        return True
+    if (abs(expr.base) <= 1) == True and ask(Q.positive(expr.exp), assumptions):
+        return True
+    if (abs(expr.base) >= 1) == True and ask(Q.negative(expr.exp), assumptions):
+        return True
+    if (abs(expr.base) >= 1) == True and exp_bounded is False:
+        return False
+    return None
+
+@FinitePredicate.register_many(exp, log)
+def _(expr, assumptions):
+    return ask(Q.finite(expr.args[0]), assumptions)
```

Comment:
This isn't valid for `log` is it?

Zero is finite but `log(0)` is not.

Reply:
If that is just preserving existing behaviour then I don't suggest to change it in this PR.